### PR TITLE
Support RTL

### LINKED ISSUE #53: Slider is broken on dir="rtl" pages.
[Issuehunt badges](https://issuehunt.io/r/brainhubeu/react-carousel/issues/53)

To reproduce that case just add dir="rtl" to <HTML> tag.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $5.00)

- [piotr-s-brainhub piotr-s-brainhub](https://issuehunt.io/u/piotr-s-brainhub) ($5.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/brainhubeu/react-carousel/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by sinchang**
- [x] `package.json:version` has been updated with `npm version patch` (or `major/minor` as appropriate)
- [x] `@brainhubeu/react-carousel` version has been updated in `docs-www/package.json` to the same which is in `package.json:version`

I have deployed the doc by Netlify on [here](https://5e27dc497b5ff6000c337118--flamboyant-payne-bea900.netlify.com/docs/examples/rtl/), you can have a quick preview.

closes #53

### CODE REVIEW

**Comment by Lukasz-pluszczewski** [comment]
Thank you for your contribution, looks good! And thanks for deploying the docs, it saves us a ton of time :+1:

The only thing I see is missing, are changes to Dots component. Are you planning to implement rtl for Dots too?

**Comment by sinchang** [comment]
@Lukasz-pluszczewski done. new preview URL on [here](https://5e27dc497b5ff6000c337118--flamboyant-payne-bea900.netlify.com/docs/examples/rtl/)

File: README.md [hunk 1]
```diff
@@ -201,6 +202,7 @@ Where:
 * `itemWidth` (*number*) determines custom width for each slide in carousel
 * `clickToChange` *boolean* indicating if clicking on a slide should trigger changing the current value
 * `centered` *boolean* indicating if the current active slide should be aligned to the center or to the left of a carousel
+* `rtl` *boolean* indicating if the carousel should be changed direction from Right to left
```
Comment (piotr-s-brainhub) [inline README.md]:
```suggestion
* `rtl` *boolean* indicating if the carousel should have direction from Right to Left
```

File: src/components/Carousel.js [hunk 2]
```diff
@@ -541,17 +543,25 @@ export default class Carousel extends Component {
...
     const trackStyles = {
-      marginLeft: `${this.getAdditionalClonesOffset()}px`,
       width: `${trackWidth}px`,
-      transform: `translateX(${transformOffset}px)`,
       transitionDuration: transitionEnabled ? `${animationSpeed}ms, ${animationSpeed}ms` : null,
     };
 
+    if (isRTL) {
+      trackStyles.marginRight = `${this.getAdditionalClonesOffset()}px`;
+      trackStyles.transform = `translateX(${transformOffset * (-1)}px)`;
```
Comment (piotr-s-brainhub) [inline src/components/Carousel.js]:
```suggestion
      trackStyles.transform = `translateX(${-transformOffset}px)`;
```

File: src/components/Carousel.js [hunk 3]
```diff
@@ -318,7 +320,7 @@ export default class Carousel extends Component {
     const { pageX } = e;
     if (this.state.dragStart !== null) {
       this.setState(previousState => ({
-        dragOffset: pageX - previousState.dragStart,
+        dragOffset: this.getProp('rtl') ? (pageX - previousState.dragStart) * -1 : pageX - previousState.dragStart,
```
Comment (piotr-s-brainhub) [inline src/components/Carousel.js]:
```suggestion
        dragOffset: this.getProp('rtl') ? previousState.dragStart - pageX : pageX - previousState.dragStart,
```

File: src/components/Carousel.js [hunk 4]
```diff
@@ -348,7 +350,7 @@ export default class Carousel extends Component {
     const { changedTouches } = e;
     if (this.state.dragStart !== null) {
       this.setState(previousState => ({
-        dragOffset: changedTouches[0].pageX - previousState.dragStart,
+        dragOffset: this.getProp('rtl') ? (changedTouches[0].pageX - previousState.dragStart) * -1 : (changedTouches[0].pageX - previousState.dragStart),
```
Comment (piotr-s-brainhub) [inline src/components/Carousel.js]:
```suggestion
        dragOffset: this.getProp('rtl') ? previousState.dragStart - changedTouches[0].pageX : changedTouches[0].pageX - previousState.dragStart,
```

**Comment by Lukasz-pluszczewski** [comment]
@sinchang Changes to Dots look good, thanks! Please consider commiting @piotr-s-brainhub's suggestions and we're ready to merge.

File: src/styles/Arrows.scss [hunk 5]
```diff
@@ -8,47 +8,71 @@ $arrowSize: 10px;
 $arrowWeight: 3px;
 
 .BrainhubCarousel__arrows {
-    position: relative;
-    padding: $clickableAreaSize/2;
-    border: none;
-    overflow: hidden;
-    outline: 0;
+  position: relative;
```
Comment (CHaNGeTe) [inline src/styles/Arrows.scss]: why no tabs?

**Comment by piotr-s-brainhub** [comment]
@sinchang 

Could you resolve the comments and conflicts?
Later you do it, there can be more conflicts.

**Comment by sinchang** [comment]
@piotr-s-brainhub sorry for the delay. it is done now.

**Comment by piotr-s-brainhub** [comment]
@sinchang 

Sorry but there are some new conflicts.

**Comment by sinchang** [comment]
@piotr-s-brainhub `publish gh pages for a given PR` action is failed. please take a look.

**Comment by piotr-s-brainhub** [comment]
@sinchang 

That's because it unfortunately works only for `brainhubeu` branches.

**Comment by piotr-s-brainhub** [comment]
@sinchang 

I opened https://github.com/brainhubeu/react-carousel/pull/221 instead.

**Pull request merged**